### PR TITLE
external: Update boost to 1.86.

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- external: Update boost to 1.86.
Adding crc module.
- Fix Boost linkage issues in release build.
Fix path of BOOST_LIBRARYDIR.

### CODE REVIEW

**Comment by Macdu** [comment]
Doesn't the CMakeFile need to be changed if you want to fix building with custom Boost?

**Comment by Zangetsu38** [comment]
there is already a pr for that

**Comment by Macdu** [comment]
I mean just build the correct variant of boost using B2 depending on if the target is debug or not, it takes 4 lines which I showed you. Redpolline's PR doesn't even do that and is nowhere near close to getting accepted from my point of view.

**Comment by Zangetsu38** [comment]
> I mean just build the correct variant of boost using B2 depending on if the target is debug or not, it takes 4 lines which I showed you. Redpolline's PR doesn't even do that and is nowhere near close to getting accepted from my point of view.

is see, but that change no broken debug build ? 🤔

File: CMakeLists.txt [hunk 1]
```diff
@@ -127,7 +127,7 @@ macro(get_boost)
 
 		  # First, try to find Boost without any hints (system Boost)
 		  if(NOT VITA3K_FORCE_CUSTOM_BOOST)
-		      find_package(Boost 1.81 COMPONENTS ${BOOST_MODULES_TO_FIND} QUIET)
+		      find_package(Boost 1.83 COMPONENTS ${BOOST_MODULES_TO_FIND} QUIET)
```
Comment (scribam) [inline CMakeLists.txt:130]: Why 1.83 is required instead of 1.81?

Comment (Zangetsu38) [reply CMakeLists.txt:130]: i just for increment minimum requiement
but for now i try fix one things...

**Comment by Zangetsu38** [comment]
Is finnaly ready, thx to @Macdu for help of fixing linking issue... stupid cmake...

File: CMakeLists.txt (hunk 1 above)
Comment (Macdu) [reply CMakeLists.txt:131]: I do agree with scribam, the minimum version should only be increased if it is required.

Comment (Zangetsu38) [reply CMakeLists.txt:131]: Ah, i have thinks is good with see all workflow using 1.83 🥲

Comment (Zangetsu38) [reply CMakeLists.txt]: I have revert it is ready

File: CMakeLists.txt [hunk 2]
```diff
@@ -138,8 +139,8 @@ macro(get_boost)
 		      set(BOOST_SOURCEDIR "${CMAKE_SOURCE_DIR}/external/boost")       # For internal use
 		      set(BOOST_INSTALLDIR "${CMAKE_BINARY_DIR}/external/boost")      # For internal use
 		      set(BOOST_ROOT "${BOOST_SOURCEDIR}")                            # find_package(Boost ...) hint
-		      set(BOOST_INCLUDEDIR "${BOOST_SOURCEDIR}")                      # find_package(Boost ...) hint
-		      set(BOOST_LIBRARYDIR "${BOOST_INSTALLDIR}/libs")                # find_package(Boost ...) hint
+		      set(BOOST_INCLUDE_DIRS "${BOOST_SOURCEDIR}")                      # find_package(Boost ...) hint
+		      set(BOOST_LIBRARY_DIRS "${BOOST_INSTALLDIR}/lib")                # find_package(Boost ...) hint
```
Comment (Macdu) [inline CMakeLists.txt:143]: Is there a reason for changing these?
According to [FindBoost](https://cmake.org/cmake/help/latest/module/FindBoost.html), the former are hints while the latter are variable set by find_package(boost) and thus shouldn't be set by the app cmake.

Comment (Zangetsu38) [reply CMakeLists.txt:143]: Ha yes... i have no see hint value... i have only see result and cache
But second fail is libs folder no exist, is lib

Comment (Zangetsu38) [reply CMakeLists.txt]: fixed.

**Review by Macdu — APPROVED** [review]

**Pull request merged**